util/gen_action_info.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python
+
+import csv
+import json
+import os
+import urllib.request
+import re
+import io
+from collections import defaultdict
+
+"""Action data is available in csv form that has 3 headers looks like:
+key,0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65
+#,Name,,Icon,ActionCategory,,Animation{Start},VFX,Animation{End},ActionTimeline{Hit},,ClassJob,BehaviourType,ClassJobLevel,IsRoleAction,Range,CanTargetSelf,CanTargetParty,CanTargetFriendly,CanTargetHostile,,,TargetArea,,,,CanTargetDead,,CastType,EffectRange,XAxisModifier,,PrimaryCost{Type},PrimaryCost{Value},SecondaryCost{Type},SecondaryCost{Value},Action{Combo},PreservesCombo,Cast<100ms>,Recast<100ms>,CooldownGroup,,MaxCharges,AttackType,Aspect,ActionProcStatus,,Status{GainSelf},UnlockLink,ClassJobCategory,,,AffectsPosition,Omen,IsPvP,,,,,,,,,,,,IsPlayerAction
+int32,str,bit&01,Image,ActionCategory,byte,ActionCastTimeline,ActionCastVFX,ActionTimeline,ActionTimeline,byte,ClassJob,byte,byte,bit&02,sbyte,bit&04,bit&08,bit&10,bit&20,bit&40,bit&80,bit&01,bit&02,bit&04,sbyte,bit&08,bit&10,byte,byte,byte,bit&20,byte,uint16,byte,Row,Action,bit&40,uint16,uint16,byte,byte,byte,AttackType,byte,ActionProcStatus,byte,Status,Row,ClassJobCategory,byte,bit&80,bit&01,Omen,bit&02,bit&04,bit&08,bit&10,bit&20,bit&40,bit&80,bit&01,bit&02,byte,bit&04,bit&08,bit&10
+0,"",False,405,0,0,0,0,0,0,0,-1,0,0,False,0,False,False,False,False,False,False,False,False,False,0,True,True,0,0,0,False,0,0,0,0,0,False,0,0,0,0,0,0,0,0,0,0,0,0,0,False,False,0,False,False,True,False,True,True,False,False,True,0,False,False,False
+===rest of data===
+
+We take the raw output and convert it into a usable JavaScript file to be imported and used within modules so we don't need to find and hardcode action data.
+
+TODO:
+Move constants into config file
+Implement argparse
+Make more universal functions more friendly for reuse
+General streamlining
+Implement column filtering on returned data
+Validate output
+Implement proper logging
+Implement proper error handling
+Review and streamline name normalization (current normalization taken from csv_util.py and coinach.py)
+Build out local file functionality
+Move table structuring from dunders to get_data_table()
+Stab feature creep/overengineering in the eye
```

Comment:
Yeah, I think most of this TODO is quite a bit of overengineering, unless I misunderstand where you're going with this.  These are all one-off scripts that get run once a release.

Reply:
I was aiming more toward a "table helper" rather than just a one off script to create action ids.  I figured since the tables are the game data, there have to be things that would be useful for other purposes.

I thought if I could create something that could be used to gather _any_ relevant data, I'd be contributing more than just one fairly simple script.

Reply:
I think I think about `csv_util` as being the "table helper", so maybe it might make sense to point some small improvements there first to handle what you need?

I think there's some limit to what it makes sense to commit into the cactbot repo.  If you want to make a Python project that extensively handles game data, that's probably far more than cactbot needs to handle and it might make sense to do that in some other repo (although feel free to fork off the cactbot utilities if they're useful to you).

I think I would want to understand what these other purposes were, before adding support for them, if that makes sense.  I am much more easily motivated by immediate use cases vs nebulous future ones.